library.go

```diff
@@ -1023,11 +1023,17 @@ func zip(thread *Thread, _ *Builtin, args Tuple, kwargs []Tuple) (Value, error)
 	for i, seq := range args {
 		it := Iterate(seq)
 		if it == nil {
+			for j := 0; j < i; j++ {
```

Comment:
Given that there are three loops, and that zip is already allocation intensive, we should probably unify all these with a single deferred call:

```
iters := make([]Iterator, cols)
defer func() {
	for _, iter := range iters {
￼ 		if iter != nil {
			iter.Done()
		}
￼	}
}()
for i, seq := range args {
```

This has the benefit of cleaning up after a recovered panic (although I don't claim to have made the code panic-safe throughout).

Reply:
In addition to this fix I would like to implement support for iterables of unknown length (as suggested in the TODO comment). I am thinking about returning an Iterator that would wrap `iters`. For this change defer won't work and `Done` will have to be called from the returned Iterator. Also after this change there will be only 2 loops. Let me create another PR so these 2 changes can be reviewed together.